### ISSUE TITLE
iOS 14 - no notes after cloning

### ISSUE THREAD

**Issue by 1001ruchka** [GitJournal/GitJournal#394]
This morning, gitjournal returned an error when trying to sync with GitHub. Restarting the app had no effect. After that, I reset the ssh keys and tried to clone the repository again. The cloning attempt failed. Cloning occurs, but there are no notes. 
I have this problem on 2 devices. iPad and iPhone on iOS 14.

In the debug error 

{"t":1610971933570,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610971933601,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610971933613,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610971933613,"l":"i","msg":"--------- App Launched ---------","stack":"null "}
{"t":1610971933613,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610971933613,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610971933613,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610971933614,"l":"i","msg":"AppSetting {onBoardingCompleted: true, collectUsageStatistics: true, collectCrashReports: true, version: 0, proMode: true, validateProMode: true, proExpirationDate: 2021-10-27T20:58:52.000Z, pseudoId: 69a4e480-e7b2-4008-914a-fbfe767e6d40, debugLogLevel: v, experimentalFs: false, experimentalMarkdownToolbar: false, experimentalGraphView: false, experimentalZeroConf: false, experimentalAccounts: false}","stack":"null "}
{"t":1610971933618,"l":"d","msg":"Analytics Collection: true","stack":"null "}
{"t":1610971933621,"l":"d","msg":"Event.Settings","stack":"null "}
{"t":1610971933621,"l":"i","msg":"Setting {noteFileNameFormat: FromTitle, journalNoteFileNameFormat: SimpleDate, yamlModifiedKey: modified, yamlCreatedKey: created, yamlTagsKey: tags, customMetaData: , yamlHeaderEnabled: false, journalEditordefaultNewNoteFolderSpec: , journalEditorSingleNote: true, defaultEditor: Markdown, defaultView: Standard, sortingField: Title, sortingOrder: asc, remoteSyncFrequency: automatic, showNoteSummary: true, folderViewHeaderType: TitleOrFileName, version: 2, markdownDefaultView: View, markdownLastUsedView: Edit, homeScreen: all_notes, imageLocationSpec: ., zenMode: false, saveTitleInH1: true, swipeToDelete: false, inlineTagPrefixes: #, emojiParser: true, folderName: journal_0, bottomMenuBar: true, confirmDelete: true, storeInternally: true, storageLocation: , sshPublicKey: true, sshPrivateKey: true}","stack":"null "}
{"t":1610971933630,"l":"i","msg":"Listing Remotes - ","stack":"null "}
{"t":1610971933630,"l":"i","msg":"Remote Name: origin","stack":"null "}
{"t":1610971933630,"l":"i","msg":"Remote URL: git@github.com:xxxxxxxx/obsidian.git","stack":"null "}
{"t":1610971933631,"l":"d","msg":"Event.RepoSynced","stack":"null "}
{"t":1610971933633,"l":"i","msg":"Checking if ProMode should be enabled. Exp: 2021-10-27T20:58:52.000Z","stack":"null "}
{"t":1610971933633,"l":"i","msg":"Not checking PurchaseInfo as exp = 2021-10-27T20:58:52.000Z and cur = 2021-01-18T12:12:13.633174Z","stack":"null "}
{"t":1610971933634,"l":"i","msg":"Old Transactions: 0","stack":"null "}
{"t":1610971933667,"l":"i","msg":"Finished loading the notes cache","stack":"null "}
{"t":1610971933671,"l":"d","msg":"Saving the NotesCache","stack":"null "}
{"t":1610971933685,"l":"i","msg":"Finished loading all the notes","stack":"null "}
{"t":1610971933765,"l":"d","msg":"Received Share with App (text): null","stack":"null "}
{"t":1610971965006,"l":"d","msg":"Event.DrawerSettings","stack":"null "}
{"t":1610971969433,"l":"i","msg":"No log file for 2021-01-17 15:12:49.432959","stack":"null "}
{"t":1610972087005,"l":"e","msg":"Got Exception","ex":"Exception: Current Branch null","stack":"package:gitjournal/core/git_repo.dart 190  GitNoteRepository.merge package:stack_trace                        StackZoneSpecification._registerUnaryCallback.<fn> <asynchronous suspension> "}
{"t":1610972087012,"l":"d","msg":"Saving the NotesCache","stack":"null "}
{"t":1610972087039,"l":"d","msg":"Synced!","stack":"null "}
{"t":1610972114372,"l":"e","msg":"Uncaught Exception","stack":"package:http/src/io_client.dart 62         IOClient.send dart:async                                 _asyncErrorWrapperHelper package:gitjournal/core/git_repo.dart 190  GitNoteRepository.merge dart:async                                 _asyncThenWrapperHelper package:gitjournal/repository.dart 208     Repository.syncNotes dart:async                                 _asyncThenWrapperHelper package:gitjournal/repository.dart 240     Repository._syncNotes package:gitjournal/repository.dart 173     new Repository._internal package:gitjournal/repository.dart 138     Repository.load dart:async                                 _asyncThenWrapperHelper package:gitjournal/app.dart 51             JournalApp.main dart:async                                 _asyncThenWrapperHelper package:gitjournal/main.dart 34            main.<fn>.<fn> package:gitjournal/main.dart 33            main.<fn>.<fn> package:stack_trace                        Chain.capture package:gitjournal/main.dart 33            main.<fn> package:gitjournal/main.dart 32            main.<fn> dart:async                                 runZonedGuarded package:gitjournal/main.dart 32            main "}
{"t":1610972116825,"l":"d","msg":"Event.RepoSynced","stack":"null "}
{"t":1610972119031,"l":"e","msg":"Got Exception","ex":"Exception: Current Branch null","stack":"package:gitjournal/core/git_repo.dart 190        GitNoteRepository.merge package:gitjournal/repository.dart 208           Repository.syncNotes package:gitjournal/widgets/sync_button.dart 101  _SyncButtonState._syncRepo <asynchronous suspension> "}
{"t":1610972119042,"l":"d","msg":"Saving the NotesCache","stack":"null "}
{"t":1610972119047,"l":"d","msg":"Synced!","stack":"null "}
{"t":1610972123690,"l":"d","msg":"Event.DrawerSettings","stack":"null "}
{"t":1610972127422,"l":"i","msg":"No log file for 2021-01-17 15:15:27.422139","stack":"null "}

**Comment by vHanda** [comment]
I'm going to revert to the previous way of cloning so that these issues go away. Let me see if I can manage to do it today.

**Comment by 1001ruchka** [comment]
{"t":1610983872256,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610983872256,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610983872256,"l":"i","msg":"--------- App Launched ---------","stack":"null "}
{"t":1610983872256,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610983872256,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610983872256,"l":"i","msg":"--------------------------------","stack":"null "}
{"t":1610983872256,"l":"i","msg":"AppSetting {onBoardingCompleted: true, collectUsageStatistics: true, collectCrashReports: true, version: 0, proMode: true, validateProMode: true, proExpirationDate: 2021-10-27T20:58:52.000Z, pseudoId: df4e4aae-4872-4b12-99b5-0f96215d8e8f, debugLogLevel: v, experimentalFs: false, experimentalMarkdownToolbar: false, experimentalGraphView: false, experimentalZeroConf: false, experimentalAccounts: false}","stack":"null "}
{"t":1610983872260,"l":"d","msg":"Analytics Collection: true","stack":"null "}
{"t":1610983872261,"l":"d","msg":"Event.Settings","stack":"null "}
{"t":1610983872261,"l":"i","msg":"Setting {noteFileNameFormat: FromTitle, journalNoteFileNameFormat: SimpleDate, yamlModifiedKey: modified, yamlCreatedKey: created, yamlTagsKey: tags, customMetaData: , yamlHeaderEnabled: false, journalEditordefaultNewNoteFolderSpec: Daily-Notes, journalEditorSingleNote: false, defaultEditor: Markdown, defaultView: Standard, sortingField: Title, sortingOrder: asc, remoteSyncFrequency: automatic, showNoteSummary: true, folderViewHeaderType: TitleOrFileName, version: 2, markdownDefaultView: View, markdownLastUsedView: Edit, homeScreen: all_notes, imageLocationSpec: Stroll, zenMode: false, saveTitleInH1: true, swipeToDelete: false, inlineTagPrefixes: #, emojiParser: true, folderName: journal_0, bottomMenuBar: true, confirmDelete: true, storeInternally: true, storageLocation: , sshPublicKey: true, sshPrivateKey: true}","stack":"null "}
{"t":1610983872265,"l":"i","msg":"Listing Remotes - ","stack":"null "}
{"t":1610983872265,"l":"i","msg":"Remote Name: origin","stack":"null "}
{"t":1610983872265,"l":"i","msg":"Remote URL: git@github.com:xxxxx/obsidian.git","stack":"null "}
{"t":1610983872266,"l":"d","msg":"Event.RepoSynced","stack":"null "}
{"t":1610983872268,"l":"i","msg":"Checking if ProMode should be enabled. Exp: 2021-10-27T20:58:52.000Z","stack":"null "}
{"t":1610983872268,"l":"i","msg":"Not checking PurchaseInfo as exp = 2021-10-27T20:58:52.000Z and cur = 2021-01-18T15:31:12.268161Z","stack":"null "}
{"t":1610983872269,"l":"i","msg":"Old Transactions: 0","stack":"null "}
{"t":1610983872285,"l":"i","msg":"Finished loading the notes cache","stack":"null "}
{"t":1610983872306,"l":"d","msg":"Saving the NotesCache","stack":"null "}
{"t":1610983872323,"l":"i","msg":"Finished loading all the notes","stack":"null "}
{"t":1610983872359,"l":"d","msg":"Received Share with App (text): null","stack":"null "}
**{"t":1610983874443,"l":"e","msg":"Git Merge Failed","ex":"GitException: reference 'refs/heads/master' not found","stack":"package:git_bindings/git_bindings.dart 55  GitRepo.merge package:stack_trace                        StackZoneSpecification._registerUnaryCallback.<fn> <asynchronous suspension> "}**
{"t":1610983874445,"l":"d","msg":"Saving the NotesCache","stack":"null "}
{"t":1610983874448,"l":"d","msg":"Synced!","stack":"null "}
{"t":1610983879704,"l":"d","msg":"Event.DrawerSettings","stack":"null "}
{"t":1610983883219,"l":"i","msg":"No log file for 2021-01-17 18:31:23.218597","stack":"null "}

**Comment by 1001ruchka** [comment]
18:59:25 Got Exception Exception: Current Branch null package:gitjournal/core/git_repo.dart 190  GitNoteRepository.merge package:stack_trace                        StackZoneSpecification._registerUnaryCallback.<fn> <asynchronous suspension>

**Comment by 1001ruchka** [comment]
Spent a few hours, but it seems to have solved the problem. First, I made a backup of the obsidian folder and deleted the repository from github. Then I re-created the repository via gitjournal, first deleting and re-installing gitjournal (apparently gitjournal keeps the repository files in the cache even if you do a reset). Next, through an external client, I copied Obsidian files  from the backup. However, every time I synced gitjournal, I came across an error:
Got Exception Exception: Current Branch null package:gitjournal/core/git_repo.dart 190 GitNoteRepository.merge package:stack_trace StackZoneSpecification._registerUnaryCallback. Then I deleted the repository, deleted gitjournal, and reinstalled it. Then, through gitjournal, I created a repository. Then I began to upload files through an external git client there. Just not all 500 files at once, as I did last time, but in batches of 20-30 files. And after each copy operation, I did synchronization in gitjournal. Thus, within 30 minutes, I copied all the files in parts and did not get a single failure.
